### PR TITLE
run android integration test on devicefarm

### PR DESCRIPTION
**Pull request by sunchunqiang**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -999,7 +1101,29 @@ workflows:
...
       - check_sdk_on_maven  
+  run_integrationtest_on_devicefarm:
+    triggers:
+      - schedule:
+        # run the job on 8pm pacific time every day
+          cron: "0 3 * * *"
+          filters:
+            branches:
+              only:
+                - devicefarm
```
Comment (minbi) [inline .circleci/config.yml]: why does this not run from the develop or master branch?

File: .circleci/config.yml [hunk 2]
```diff
@@ -999,7 +1101,29 @@ workflows:
                 - master      
     jobs:
       - check_sdk_on_maven  
+  run_integrationtest_on_devicefarm:
+    triggers:
+      - schedule:
+        # run the job on 8pm pacific time every day
```
Comment (kvasukib) [inline .circleci/config.yml]: 8PM PST

File: .circleci/config.yml [hunk 3]
```diff
@@ -999,7 +1101,29 @@ workflows:
...
+            branches:
+              only:
+                - master      
+    jobs:
+      - run_integrationtest_on_devicefarm  
 
+  check_testresult_on_devicefarm:
+    triggers:
+      - schedule:
+        # run the job on 8am pacific time every day
```
Comment (kvasukib) [inline .circleci/config.yml]: 8AM PST

File: CircleciScripts/get_testresult_on_devicefarm.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+with open(configurefile, 'r') as jsonfile:
+    jsonstring = jsonfile.read()
+configure =  demjson.decode(jsonstring)
+
+project_arn =  configure['project_arn']
+project_runlink = configure['project_runlink']
+
+ 
+# session = boto3.session.Session(profile_name='default')
```
Comment (kvasukib) [inline CircleciScripts/get_testresult_on_devicefarm.py]: Remove commented code

File: CircleciScripts/get_testresult_on_devicefarm.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        status = run['status']
+        if status != 'COMPLETED' :
+            continue
+        testname =  run['name']
+        if not testname.endswith(tag):
+            continue
+        totals += 1
+        if run['result'] == "PASSED":
+            passeds += 1
+        # testname = testname[:-len(tailtag)]
```
Comment (kvasukib) [inline CircleciScripts/get_testresult_on_devicefarm.py]: Remove commented code

File: CircleciScripts/get_testresult_on_devicefarm.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        total = int(run['counters']['total'])  ,
+        passed = int(run['counters']['passed'])   ,
+        failed = run['counters']['failed'] ,
+        warned = run['counters']['warned'] ,
+        skipped = run['counters']['skipped'] ,    
+        errored = run['counters']['errored'] ,  
+        totalminutes = run['deviceMinutes']['total']
+        )
+
+        # print(result)
```
Comment (kvasukib) [inline CircleciScripts/get_testresult_on_devicefarm.py]: Remove commented code

File: CircleciScripts/runtest_on_devicefarm.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+    #schedule an run for the test module
+    response = client.schedule_run(
+        projectArn = project_arn,
+        appArn = dummy_apk_arn,
+        devicePoolArn= devicepool_arn,
+        name='{0}-{1}'.format(name, tag),
+        test={
+            'type': 'INSTRUMENTATION',
+            'testPackageArn': uploadarn,
+            # 'testSpecArn': testspec_arn
```
Comment (kvasukib) [inline CircleciScripts/runtest_on_devicefarm.py]: remove commented code

File: CircleciScripts/send_email.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        content = markdowner.convert(content)
+    if htmlformat :
+        message['Body']['Html'] =  {
+                    'Data': content 
+                }
+    else :
+        message['Body']['Text'] =  {
+                    'Data': content 
+                }    
+    # session = boto3.session.Session(profile_name='default')
```
Comment (kvasukib) [inline CircleciScripts/send_email.py]: remove commented code

File: CircleciScripts/get_testresult_on_devicefarm.py (hunk 4 above)
Comment (sunchunqiang) [reply CircleciScripts/get_testresult_on_devicefarm.py]: done

File: CircleciScripts/send_email.py (hunk 8 above)
Comment (sunchunqiang) [reply CircleciScripts/send_email.py]: done

File: CircleciScripts/runtest_on_devicefarm.py (hunk 7 above)
Comment (sunchunqiang) [reply CircleciScripts/runtest_on_devicefarm.py]: done

File: CircleciScripts/get_testresult_on_devicefarm.py (hunk 6 above)
Comment (sunchunqiang) [reply CircleciScripts/get_testresult_on_devicefarm.py]: done

File: CircleciScripts/get_testresult_on_devicefarm.py (hunk 5 above)
Comment (sunchunqiang) [reply CircleciScripts/get_testresult_on_devicefarm.py]: done

File: CircleciScripts/get_testresult_on_devicefarm.py (hunk 4 above)
Comment (sunchunqiang) [reply CircleciScripts/get_testresult_on_devicefarm.py]: done

File: .circleci/config.yml (hunk 3 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 2 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 1 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: done

**Pull request merged**